### PR TITLE
Update schema for VMSS

### PR DESCRIPTION
**Pull request by DeepakRajendranMsft**

### CODE REVIEW

File: schemas/2015-08-01/Microsoft.Compute.json [hunk 1]
```diff
@@ -802,13 +802,21 @@
         "name": {
           "type": "string"
         },
-        "subnet": {
-          "$ref": "#/definitions/id"
-        },
-        "loadBalancerBackendAddressPools": {
-          "type": "array",
-          "items": {
+		"properties" : {
```
Comment (daschult) [inline schemas/2015-08-01/Microsoft.Compute.json]: Did you mean to add this extra layer of properties? This means that ipConfiguration went from matching this:
{
  "name": "blah",
  "subnet": { ... },
  "loadBalancerBackendAddressPools": [ ... ],
}
to
{
  "name": "blah",
  "properties": { ... }
}
In fact, I'm not even sure that this is a valid JSON schema. Do you have any tests written for this definition?

Comment (daschult) [inline schemas/2015-08-01/Microsoft.Compute.json]: I just saw the second part of your pull request. Just for testing purposes, try adding "additionalProperties": false line just before the "name" property of the ipConfiguration definition, and then run the tests again. I think you'll find that they fail.

File: tests/2015-08-01/Microsoft.Compute.tests.json [hunk 2]
```diff
@@ -242,7 +242,22 @@
...
+        "properties":{
+          "subnet" : {
+            "id": "resourceId" 
+          },
+          "loadBalancerBackendAddressPools": [
+            {
+              "id": "backendaddresspoolId"
+            }
+          ],
+          "loadBalancerInboundNatPools": [
```
Comment (daschult) [inline tests/2015-08-01/Microsoft.Compute.tests.json]: Is it valid for these arrays to be empty? What about having two elements in the arrays? Can you add a test for those conditions?

**Comment by daschult** [comment]
Looks good to me.

**Comment by TianoMS** [comment]
Looks good to me.

**Comment by DeepakRajendranMsft** [comment]
@TianoMS @daschult who can help me merge?

**Comment by daschult** [comment]
I can't do it. Tian will have to do it.

**Pull request merged**